desc/objectives/_neoclassical.py

```diff
@@ -0,0 +1,253 @@
+                L=eq.L_grid, M=eq.M_grid, N=eq.N_grid, NFP=eq.NFP
+            )
+        if self._grid_1dr is None:
+            self._grid_1dr = self._grid_0d
+        self._dim_f = self._grid_1dr.num_rho
+        self._rho = self._grid_1dr.compress(self._grid_1dr.nodes[:, 0])
+        self._target, self._bounds = _parse_callable_target_bounds(
+            self._target, self._bounds, self._rho
+        )
+
+        timer = Timer()
+        if verbose > 0:
+            print("Precomputing transforms")
+        timer.start("Precomputing transforms")
+
+        if self._grid_1dr == self._grid_0d:
+            self._constants["transforms_0d"] = get_transforms(
+                self._keys_0d + self._keys_1dr, eq, self._grid_0d, use_jit
+            )
+            self._constants["profiles_0d"] = get_profiles(
+                self._keys_0d + self._keys_1dr, eq, self._grid_0d, use_jit
+            )
+            self._constants["transforms_1dr"] = self._constants["transforms_0d"]
+            self._constants["profiles_1dr"] = self._constants["profiles_0d"]
+        else:
+            self._constants["transforms_0d"] = get_transforms(
+                self._keys_0d, eq, self._grid_0d, use_jit
+            )
+            self._constants["profiles_0d"] = get_profiles(
+                self._keys_0d, eq, self._grid_0d, use_jit
+            )
+            self._constants["transforms_1dr"] = get_transforms(
+                self._keys_1dr, eq, self._grid_1dr, use_jit
+            )
+            self._constants["profiles_1dr"] = get_profiles(
+                self._keys_1dr, eq, self._grid_1dr, use_jit
+            )
+
+        timer.stop("Precomputing transforms")
+        if verbose > 1:
+            timer.disp("Precomputing transforms")
+
+        super().build(use_jit=use_jit, verbose=verbose)
+
+    def compute(self, params, constants=None):
+        """Compute the effective ripple.
+
+        Parameters
+        ----------
+        params : dict
+            Dictionary of equilibrium degrees of freedom, eg Equilibrium.params_dict
+        constants : dict
+            Dictionary of constant data, eg transforms, profiles etc. Defaults to
+            self.constants
+
+        Returns
+        -------
+        effective_ripple : ndarray
+
+        """
+        if constants is None:
+            constants = self.constants
+        eq = self.things[0]
+        data = compute_fun(
+            eq,
+            self._keys_0d,
+            params,
+            constants["transforms_0d"],
+            constants["profiles_0d"],
+        )
+        data = compute_fun(
+            eq,
+            self._keys_1dr,
+            params,
+            constants["transforms_1dr"],
+            constants["profiles_1dr"],
+            data={key: data[key] for key in data if eq.is_0d(key)},
+        )
+        iota = self._grid_1dr.compress(data["iota"])
+        iota_r = self._grid_1dr.compress(data["iota_r"])
+        grid = eq.rtz_grid(
```

Comment:
i think you need to be passing `params` in here somehow so that the coordinate mapping uses the new values for `L_lmn` etc. Right now this will always use the values of the initial equilibrium